src/base/settingsstorage.cpp

```diff
@@ -26,7 +26,9 @@
  * but you are not obligated to do so. If you do not wish to do so, delete this
  * exception statement from your version.
  */
+#include "settingsstorage.h"
```

Comment:
This "good technique" is almost entirely based on opinions, rather than facts, references, or specific expertise.
If some header is incomplete and this leads to error it's easier to find it in any case.  Moreover, there is no reason to change it in that file, which obviously compiles without such errors.